storm-core/src/jvm/org/apache/storm/daemon/supervisor/SyncSupervisorEvent.java

```diff
@@ -0,0 +1,626 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.daemon.supervisor;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.storm.Config;
+import org.apache.storm.blobstore.BlobStore;
+import org.apache.storm.blobstore.ClientBlobStore;
+import org.apache.storm.cluster.IStateStorage;
+import org.apache.storm.cluster.IStormClusterState;
+import org.apache.storm.event.EventManager;
+import org.apache.storm.generated.*;
+import org.apache.storm.localizer.LocalResource;
+import org.apache.storm.localizer.LocalizedResource;
+import org.apache.storm.localizer.Localizer;
+import org.apache.storm.utils.*;
+import org.apache.thrift.transport.TTransportException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.JarURLConnection;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class SyncSupervisorEvent implements Runnable {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SyncSupervisorEvent.class);
+
+    private EventManager syncSupEventManager;
+    private EventManager syncProcessManager;
+    private IStormClusterState stormClusterState;
+    private LocalState localState;
+    private SyncProcessEvent syncProcesses;
+    private SupervisorData supervisorData;
+
+    public SyncSupervisorEvent(SupervisorData supervisorData, SyncProcessEvent syncProcesses, EventManager syncSupEventManager,
+            EventManager syncProcessManager) {
+
+        this.syncProcesses = syncProcesses;
+        this.syncSupEventManager = syncSupEventManager;
+        this.syncProcessManager = syncProcessManager;
+        this.stormClusterState = supervisorData.getStormClusterState();
+        this.localState = supervisorData.getLocalState();
+        this.supervisorData = supervisorData;
+    }
+
+    @Override
+    public void run() {
+        try {
+            Map conf = supervisorData.getConf();
+            Runnable syncCallback = new EventManagerPushCallback(this, syncSupEventManager);
+            List<String> stormIds = stormClusterState.assignments(syncCallback);
+            Map<String, Map<String, Object>> assignmentsSnapshot =
+                    getAssignmentsSnapshot(stormClusterState, stormIds, supervisorData.getAssignmentVersions().get(), syncCallback);
+            Map<String, List<ProfileRequest>> stormIdToProfilerActions = getProfileActions(stormClusterState, stormIds);
```

Comment:
here stormIdToProfilerActions contain all storm topology, suggest put it later and filter some topology,  only contain the topology which will run on current supervisor.